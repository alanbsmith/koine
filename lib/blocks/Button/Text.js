import styled from 'styled-components';
import { rem } from 'polished';

const Text = styled.span`
  font-size: ${rem(12)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
