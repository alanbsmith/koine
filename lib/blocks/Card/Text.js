import styled from 'styled-components';
import { rem } from 'polished';

import P from '../../elements/P';

const Text = styled(P)`
  width: ${rem(240)};
`;

export default Text;
