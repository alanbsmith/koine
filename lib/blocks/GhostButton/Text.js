import styled from 'styled-components';
import gridScale from '../../utils/gridScale';

const Text = styled.span`
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
`;

export default Text;
