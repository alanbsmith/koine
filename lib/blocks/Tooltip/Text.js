import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { fontStyles, fontWeights } from '../../modifiers';
import gridScale from '../../utils/gridScale';

const modifierConfig = {
  ...fontStyles,
  ...fontWeights,
};

const Text = styled.span`
  border-radius: ${props => props.theme.dimensions.borderRadius};
  color: ${props => props.theme.colors.base.chrome000};
  font-size: ${props => gridScale(props, 1.5)};
  font-weight: ${props => props.theme.fontWeights.light};
  text-transform: uppercase;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Text;
