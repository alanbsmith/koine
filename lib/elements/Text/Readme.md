```jsx
<div>
  <Text modifiers={['capitalize']}>hello world</Text>
</div>
```
