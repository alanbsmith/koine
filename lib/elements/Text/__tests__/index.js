import React from 'react';
import Text from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';

describe('Paragraph Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Text>A paragraph</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the extraSmall modifier', () => {
    const tree = renderWithTheme(<Text modifiers={['extraSmall']}>A paragraph</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the small modifier', () => {
    const tree = renderWithTheme(<Text modifiers={['small']}>A paragraph</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the large modifier', () => {
    const tree = renderWithTheme(<Text modifiers={['large']}>A paragraph</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
