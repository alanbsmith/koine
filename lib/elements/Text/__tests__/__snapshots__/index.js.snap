// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Element adds the extraSmall modifier 1`] = `
.c0 {
  display: inline-block;
  font-size: 1rem;
  margin: 0;
  font-size: 0.75rem;
}

<span
  className="c0"
>
  A paragraph
</span>
`;

exports[`Paragraph Element adds the large modifier 1`] = `
.c0 {
  display: inline-block;
  font-size: 1rem;
  margin: 0;
  font-size: 1.125rem;
}

<span
  className="c0"
>
  A paragraph
</span>
`;

exports[`Paragraph Element adds the small modifier 1`] = `
.c0 {
  display: inline-block;
  font-size: 1rem;
  margin: 0;
  font-size: 0.875rem;
}

<span
  className="c0"
>
  A paragraph
</span>
`;

exports[`Paragraph Element renders correctly 1`] = `
.c0 {
  display: inline-block;
  font-size: 1rem;
  margin: 0;
}

<span
  className="c0"
>
  A paragraph
</span>
`;
